adr/adr.001.folder structure/adr.001.folder structure.md

```diff
@@ -0,0 +1,257 @@
+# ADR-001: Feature-Based Folder Structure
+
+## Status
+Proposed
+
+## Date
+- **Proposed**: 2024-10-16
+- **Decided**: 
+- **Work Began**: 
+- **Work Completed**: 
+
+## Context
+
+Our current codebase is organized mainly by the type of files and the framework components, such as `components`, `data`, `hooks`, `screens`, `utils`, etc. While this structure groups similar types of files together, it scatters related functionality across multiple directories. This can make navigation and maintenance challenging, especially as the codebase grows.
+
+We are increasingly relying on AI-assisted tooling for development. Having code that changes together located in different parts of the codebase reduces the effectiveness of such tools. Packaging by feature would enhance productivity by allowing developers (and AI tools) to focus on a single directory containing all relevant code for a feature.
+
+## Proposal
+
+We will reorganize the codebase to follow a feature-based folder structure. This means that all code related to a specific feature will be grouped together under a common directory within a `features/` folder. Shared components and utilities will be placed in a `shared/` directory.
+
+We have identified the following top-level features:
+
+- Accounts
+- Conversations
+- Groups
+- Transactions
+- Profiles
+- Search
+- Notifications
+- Frames
+- Settings
+- Recommendations
+
+Each feature directory may contain any of the following directories:
```

Comment:
these will evolve as our architecture becomes more mature/consolidated

Reply:
Is onboarding a feature?

Reply:
I think it should be personally

Reply:
Onboarding as in "this is the first time you've ever opened our app and have no accounts setup" - I believe so. 

We need to be careful not to let account linking "leak" into onboarding though! - as `Account(s)` are a much more important feature.